### DISCUSSION TITLE
submitting the form if there is an error in the form

### DISCUSSION THREAD

**Discussion by Hodosov** [react-hook-form/react-hook-form#11825] [General]
this is an example from the official documentation
https://codesandbox.io/p/sandbox/seterror-single-field-errors-79wcr9?file=%2Fsrc%2FApp.js

please explain why this code works?

in my opinion, if there is an error, the form should not be submitted. Correct me if I am wrong. or is there a way to do this using a library?

**Comment by koobawuah** [comment]
I just encountered this and I had the same opinion. If I have a custom error that is based on my specified condition, I would think if the setError() is used on the useForm() objects then the form would not submit because there is an error. If you find an explanation or a better solution, please let me know
